Core/include/Acts/Seeding/SeedFinderOrthogonal.ipp

```diff
@@ -684,23 +684,22 @@ void SeedFinderOrthogonal<external_spacepoint_t>::processFromMiddleSP(
 
 template <typename external_spacepoint_t>
 auto SeedFinderOrthogonal<external_spacepoint_t>::createTree(
-    const std::vector<const external_spacepoint_t *> &spacePoints) const
-    -> tree_t {
+    const std::vector<external_spacepoint_t> &spacePoints) const -> tree_t {
   std::vector<typename tree_t::pair_t> points;
 
   /*
    * For every input point, we create a coordinate-pointer pair, which we then
    * linearly pass to the k-d tree constructor. That constructor will take
    * care of sorting the pairs and splitting the space.
    */
-  for (const external_spacepoint_t *sp : spacePoints) {
+  for (const external_spacepoint_t &sp : spacePoints) {
     typename tree_t::coordinate_t point;
 
-    point[DimPhi] = sp->phi();
-    point[DimR] = sp->radius();
-    point[DimZ] = sp->z();
+    point[DimPhi] = sp.phi();
+    point[DimR] = sp.radius();
+    point[DimZ] = sp.z();
 
-    points.emplace_back(point, sp);
+    points.emplace_back(point, &sp);
```

Comment:
I agree with what you say, but it would involve a major change in several parts of the ACTS projects. These changes would be out of the scope of this PR and I'd like to follow that up in another PR. 
For the time being we can assure that `spacePoints` does not get modified nor goes out of scope.

Can you open an issue about this and assign it to me?